OneCommunityGlobal/HighestGoodNetworkApp
# Anirudh SK (Old PR Conflicts fix) - Jay - PERMISSION: Modify Badge Amount Permission

# Description
This PR enables the user with the Modify Badge Amount Permission to be able to edit other users' badge counts on their profile pages. 

## Related PRS (if any):
This frontend PR is related to the [#690](https://github.com/OneCommunityGlobal/HGNRest/pull/690) backend PR.

…

## Main changes explained:
- Added a new permission lable for Modify Badge Amount.
- Added conditional logic so that delete and Featured checkbox are disabled for users with only the modify badge count permission.
- Added conditional logic so that the user is not able to lower the badge count below 1 so that he/she is not able to delete a badge on accident.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as owner user
5. go to Other Links -> Permissions Management -> Manage User Permissions -> Choose a user without badge management permissions such as volunteer or core team -> add Modify Badge Amount 
6. verify that user is now able to see featured button and is able to manipulate the badge count.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/99edf90e-b437-4c28-bd02-28e9fb71c1f7


## Note:
**Unmute the video**
